pinot-broker/src/test/java/org/apache/pinot/broker/routing/instanceselector/InstanceSelectorTest.java

```diff
@@ -581,6 +585,132 @@ public void testInstanceSelector() {
     assertTrue(selectionResult.getUnavailableSegments().isEmpty());
   }
 
+  @Test
+  public void testReplicaGroupInstanceSelectorNumReplicaGroups() {
+    String offlineTableName = "testTable_OFFLINE";
+    BrokerMetrics brokerMetrics = mock(BrokerMetrics.class);
+    BrokerRequest brokerRequest = mock(BrokerRequest.class);
+    PinotQuery pinotQuery = mock(PinotQuery.class);
+    Map<String, String> queryOptions = new HashMap<>();
+    queryOptions.put("numReplicaGroups", "2");
```

Comment:
Can you also cover the case with `numReplicaGroups=4`?

Reply:
added